### ISSUE TITLE
Create the one-/two-/three-dimensional line groups with the help of this package based on the Orbifold/Schönflies/Conway/Coxeter notations.

### ISSUE THREAD

**Issue by hongyi-zhao** [rgugliel/CoxIter#27]
Hi here,

I noticed some examples of line groups given [here](https://en.wikipedia.org/wiki/Line_group), as shown below:

![image](https://user-images.githubusercontent.com/11155854/209745728-057a91b3-f242-4cf3-bfa7-9668e3467e19.png)
![image](https://user-images.githubusercontent.com/11155854/209745772-9e524466-3070-457d-b825-b4a847f8ab3f.png)
![image](https://user-images.githubusercontent.com/11155854/209745812-eafa7bea-c08e-4819-8bd8-8808bf978bac.png)

I'm not sure whether it's possible for me to create these one-/two-/three-dimensional line groups with the help of **CoxIter** package based on the `Orbifold/Schönflies/Conway/Coxeter` notations given above?

Also see the related discussion [here](https://github.com/gap-packages/hap/issues/100).

Any tips will be appreciated.

Regards,
Zhao

**Comment by rgugliel** [comment]
Dear @hongyi-zhao ,
Any Coxeter graph can be fed to CoxIter. Now, if you want to get meaningful results out of this, the specified group needs to be an hyperbolic Coxeter group. The reason for that is that CoxIter relies on fundamental results around hyperbolic Coxeter groups (see https://www.cambridge.org/core/journals/lms-journal-of-computation-and-mathematics/article/coxiter-computing-invariants-of-hyperbolic-coxeter-groups/A117FD150E108CACCACF9EF00C61A23C).
Best,
Raf

**Comment by hongyi-zhao** [comment]
Thank you very much for your reply and explanations. My other related puzzles:

1. How should I encode, a.k.a., adopting appropriate formats based on the original Coxeter graph given on the above website, so that it can be recognized by CoxIter? 
2. How do I know if the group in question is a hyperbolic Coxeter group before I input it into CoxIter?
3. Can this tool produce the output for [GAP](https://www.gap-system.org/)?

**Comment by rgugliel** [comment]
1. I'd recommend reading CoxIter documentation: https://rgugliel.github.io/CoxIter/page_cox_iter.html
2. This can be computed from the Gram matrix (see the introduction of the article: https://www.cambridge.org/core/journals/lms-journal-of-computation-and-mathematics/article/coxiter-computing-invariants-of-hyperbolic-coxeter-groups )/A117FD150E108CACCACF9EF00C61A23C)
3. The sole purpose of CoxIter is to compute invariants so there is no real output of CoxIter that can be fed into GAP. You could do the opposite: trying to use CoxIter from within GAP (see: https://github.com/rgugliel/CoxIterGAP). Please note that CoxIterGAP is slightly outdated and would benefit from some love before being used again.

Best,

Rafael

**Comment by hongyi-zhao** [comment]
> 3\. Please note that CoxIterGAP is slightly outdated and would benefit from some love before being used again.

It really hasn't been updated in years. Don't you have plans to update it?

**Comment by rgugliel** [comment]
> > 3. Please note that CoxIterGAP is slightly outdated and would benefit from some love before being used again.
> 
> It really hasn't been updated in years. Don't you have plans to update it?

Not really, and for several reasons. First, I am not in academia anymore and everything I do around CoxIter is on my free time. Second, I never had the impression that CoxIterGAP was really used, which diminish my motivation to invest time there. Also, both CoxIter and CoxIterGAP are open source, which means that anyone can improve them by submitting pull requests :)
I may consider updating CoxIterGAP if I get evidence that it is used.
